spyderlib/config/user.py

```diff
@@ -117,8 +118,27 @@ def _write_file(fname):
                 raise(e)
 
     def filename(self):
+        """Defines the name of the configuration file to use."""
+        self._filename = getattr(self, '_filename', None)
+        self._root_path = getattr(self, '_root_path', None)

```

Comment:
Hmmmm I think I did that and it was not working... I can double check (was a looooooooooooooong time ago)